### PR TITLE
cmake: Fix crash when no C++ compiler is not installed (fixes #6559)

### LINKED ISSUE #6559: meson claims Unknown compiler(s): ['c++', 'g++', 'clang++', 'pgc++', 'icpc']
This is kinda weird: I have a very small dummy meson.build which errors out with:

> meson.build:7:0: ERROR: Unknown compiler(s): ['c++', 'g++', 'clang++', 'pgc++', 'icpc']

Whereas it's correct I don't have any of those compilers on my system, there is also no indication in my meson.build that this would be needed.

The sample meson.build is:

```
project('sample',
	'c',
	version: '1.0.0',
	meson_version: '>= 0.47',
)

dep = dependency('doesnotexist', required: false)
```

### PR DESCRIPTION
**Pull request by mensinda**
No test, since this would require the absence of a C++ compiler.

### CODE REVIEW

File: mesonbuild/cmake/executor.py [hunk 1]
```diff
@@ -23,6 +23,7 @@
 import os
 import shutil
 import ctypes
+import textwrap
```
Comment (jon-turney) [inline mesonbuild/cmake/executor.py]: Is this reindentation also part of the fix?

Comment (mensinda) [reply mesonbuild/cmake/executor.py]: No, should I make a seperate PR/commit?

**Comment by nirbheek** [comment]
You should be able to emulate the absence of a C++ compiler by setting the `CPP` env var to `false` or something like that, no?

File: mesonbuild/cmake/executor.py (hunk 1 above)
Comment (nirbheek) [reply mesonbuild/cmake/executor.py]: Next time, yes, I'll just merge this for now :)

**Pull request merged**